docs/app/docs/utils/code-loader.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { of as observableOf } from 'rxjs/observable/of';
+import { publishReplay } from 'rxjs/operators/publishReplay';
+import { refCount } from 'rxjs/operators/refCount';
+import { catchError } from 'rxjs/operators/catchError';
+
+@Injectable()
+export class CodeLoaderService {
+
+  /**
+   * Contains cached files by url.
+   * */
+  private cache: Map<string, Observable<string>> = new Map();
+
+  constructor(private http: HttpClient) {
+  }
+
+  load(path: string): Observable<string> {
+    const url = this.buildFilePath(path);
+    const cached = this.cache.get(url);
+
+    if (cached) {
+      return cached;
+    }
+
+    return this.buildRequest(url);
```

Comment:
`return cached ? cached : this.buildRequest(url);`

Reply:
done